core/ustring.cpp

```diff
@@ -1337,7 +1337,17 @@ String String::num_scientific(double p_num) {
 	char buf[256];
 
 #if defined(__GNUC__) || defined(_MSC_VER)
+
+#if defined(_TWO_DIGIT_EXPONENT)
```

Comment:
Is it guaranteed that `_TWO_DIGIT_EXPONENT` is only defined by MinGW and MSVC < 2015?

Reply:
Googling gives a good probability, but no guarantees. I guess to be more correct we could check for the correct msvc version || mingw?

Reply:
Yeah, though that should likely be something like:
```
#if defined(__MINGW32__) || (defined(_MSC_VER) && _MSC_VER < 1900)
```
Might also be worth checking if recent MinGW versions fix that / if they have plans to change it eventually, since it seems they set the default to `_THREE_DIGIT_EXPONENT` for compatibility with the Windows API, which seems not longer to be relevant if MSVC changed that. (But MinGW often has a few years lag, understandably.)

Reply:
From the conversations I've seen, it looks like they only will maintain compatibility with the msvcrt that ships with Windows, the runtime that ships with vc++ 2015 etc is an add-on redistributable package and the goal of 'run without dependencies' on window would conflict. 

I also wondered if there is a good place to set this behavior on program initialization instead of this specific location? That might clean up this section at least.